dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -832,29 +866,42 @@ boolean handleOutliers(ThroughputDistribution distribution, double throughput) {
             Integer priorKey = Integer.valueOf(poolSize);
             int smallerPools = 0;
             while (true) {
-                int distance = 0;
                 // stop if we run out of data
                 if (shrinkKey == null)
                     break;
                 priorKey = shrinkKey;
                 shrinkKey = threadStats.lowerKey(shrinkKey);
+                int distance = 0;
+                boolean pruned = false;
+                boolean inHangBuffer = true;
                 // discard invalid data (old or out-of-range) found in comparison stats
                 ThroughputDistribution priorStats = threadStats.get(priorKey);
                 distance = poolSize - priorKey.intValue();
                 if (pruneData(priorStats, forecast, distance, downwardCompareSpan)) {
-                    threadStats.remove(priorKey);
-                } else {
+                    if (priorKey > hangBufferPoolSize) {
```

Comment:
Can this if statement be put first before the pruneData condition in the previous if statement?

Reply:
ya, makes sense ... moved the inHangBuffer assignment also, so now it looks like:

                if (priorKey > hangBufferPoolSize) {
                    inHangBuffer = false;
                    if (pruneData(priorStats, forecast, distance, downwardCompareSpan)) {
                        threadStats.remove(priorKey);
                        pruned = true;
                    }
                }